src/lib.rs

```diff
@@ -21,18 +21,12 @@ pub fn plugin_registrar(reg: &mut Registry) {
     reg.register_macro("speculate", expand_speculate);
 }
 
-fn expand_speculate(cx: &mut ExtCtxt, _sp: Span, tokens: &[TokenTree]) -> Box<MacResult + 'static> {
+fn expand_speculate(cx: &mut ExtCtxt, sp: Span, tokens: &[TokenTree]) -> Box<MacResult + 'static> {
+    let mod_name = format!("speculate #{}", sp.lo().0);
+    
     let mut parser = stream_to_parser(cx.parse_sess(), tokens.iter().cloned().collect());
-    let block = parser::parse(&mut parser);
+    let block = parser::parse(&mod_name, &mut parser);
     let item = block.generate(cx, None);
 
-    let module = quote_item!(cx,
-                             #[allow(non_snake_case)]
-                             mod speculate {
-                                 #[allow(unused_imports)]
-                                 use super::*;
```

Comment:
I wonder why this isn't needed anymore.

Reply:
I tried a few things with top-level imports from `std` as well as some functoons, and `use super::*;` didn't change anything. I still pushed the changes that add `import super::*` everywhere.